de_web_plugin.cpp

```diff
@@ -4805,6 +4835,61 @@ void DeRestPluginPrivate::checkSensorButtonEvent(Sensor *sensor, const deCONZ::A
                     }
 
                 }
+                else if (ind.clusterId() == COLOR_CLUSTER_ID && 
+                        ((zclFrame.commandId() == 0x05 ) || // Step Saturation
+                        (zclFrame.commandId() == 0x4C )) && // Step Color Temperature
+                        sensor->modelId().startsWith(QLatin1String("LXEK-5")))
+                {
+                    ok = false;
+                    if (zclFrame.payload().size() >= 1 && buttonMap.zclParam0 == zclFrame.payload().at(0)) // direction
+                    {
+                        sensor->previousDirection = zclFrame.payload().at(0);
+                        ok = true;
+                    }
+                    if (zclFrame.payload().size() >= 3)
+                    {
+                        //First value is a quint8 for step mode : 0x00 UP and 0x01 DOWN
+                        //The second one is step size
+                        //The third value is a quint16 for transition
+                        if (sensor->modelId().startsWith(QLatin1String("LXEK-5"))) // ADEO Lexman Télécommande (Leroy Merlin)
+                        {
+                            //need to use stepsize too for this device
+                            ok = true;
+                            quint16 param = (quint16)zclFrame.payload().at(0) & 0xff;
+                            param <<= 8;
+                            param |= (quint16)zclFrame.payload().at(1) & 0xff;
+                            if (buttonMap.zclParam0 != param)
+                            {
+                                ok = false;
+                            }
+                        }
+                    }
+                }
+                else if (ind.clusterId() == COLOR_CLUSTER_ID &&
+                        (zclFrame.commandId() == 0x02 || // Step Hue
+                         zclFrame.commandId() == 0x4C || // Step Color Temperature
```

Comment:
There is already a handler for 0x4C in line 4785 I think for this command the if will never trigger?

Reply:
Good question

```
                else if (ind.clusterId() == COLOR_CLUSTER_ID &&
                         (zclFrame.commandId() == 0x4b && zclFrame.payload().size() >= 7))  // move color temperature
                {
```

Can ask him again, but perhaps the zclFrame.payload().size()  was < 7 ?

Reply:
Will ask, it s strange.

Reply:
Ok zclFrame.commandId() == 0x4b is used at least on 3 check ...

Reply:
And now ?

Reply:
One 0x4b also checks the model id which should be safe, the once with the payload size should also hit if it is a standard command.

Reply:
For the ZBT-CCTSwitch-D0001 ?

Yep I think it s standard, but the value used in the json file can be specific, I can remove this part, but we can have to change the json too.

Reply:
I'm not sure if I completely understand the handling but can we remove the ZBT-CCTSwitch-D0001 code?
I think we need to check the related issues if the switch sends normal *Move Color Temperature* commands which can be handled by the button map.

Reply:
Yep I think this part can be handled by the button map

  ```
              else if (ind.clusterId() == COLOR_CLUSTER_ID &&
                         zclFrame.commandId() == 0x0a && zclFrame.payload().size() >= 2 && // Move to Color Temperature
                         sensor->modelId().startsWith(QLatin1String("ZBT-Remote-ALL-RGBW")))
                {
                    quint16 ct;
                    QDataStream stream(zclFrame.payload());
                    stream.setByteOrder(QDataStream::LittleEndian);
                    stream >> ct;

                    if      (sensor->previousCt < ct) ok = buttonMap.zclParam0 == 0; // CtUp
                    else if (sensor->previousCt > ct) ok = buttonMap.zclParam0 == 1; // CtDown
                    else if (ct == 370)               ok = buttonMap.zclParam0 == 0; // DimUp
                    else if (ct == 153)               ok = buttonMap.zclParam0 == 1; // DimDown
                    if (ok)
                    {
                        sensor->previousCt = ct;
                    }

                }
                else if (ind.clusterId() == COLOR_CLUSTER_ID &&
                         zclFrame.commandId() == 0x0a && zclFrame.payload().size() >= 2 && // Move to Color Temperature
                         sensor->modelId().startsWith(QLatin1String("ZBT-CCTSwitch-D0001")))
                {
                        if (buttonMap.zclParam0 != pl0)
                        {
                            ok = false;
                            pl0 = zclFrame.payload().isEmpty() ? 0 : zclFrame.payload().at(1);
                        }
                        //ignore the command if previous was button4
                        if (sensor->previousCommandId == 0x04)
                        {
                            ok = false;
                        }
                }
```

But those device are relay strange ^^, another part with same command but not handled same.

I think it will be less dangerous making a PR just to clean them, and asking for some tests.